projects/plugins/boost/app/modules/critical-css/class-critical-css.php

```diff
@@ -146,6 +146,8 @@ protected function on_initialize() {
 
 		if ( $this->should_display_critical_css() ) {
 			Admin_Bar_Css_Compat::init();
+			add_filter( 'jetpack_boost_async_style', array( $this, 'asynchronize_stylesheets_media' ), 10, 3 );
```

Comment:
I did not get that. Do you mean if the instance of `Critical_CSS` is accessible from outside Jetpack Boost plugin?

Reply:
Yeah - I haven't looked at the code. Is the instance accessible?

Reply:
It is not accessible currently.

Reply:
If a method isn't accessible from the outside, it can't be removed using `remove_filter`, and we'll lose the behavior I mentioned previously here:

https://github.com/Automattic/jetpack/pull/21665#issuecomment-966318908
> [..] as long as we make sure that `jetpack_boost_only_screen_media` is a function/method accessible from anywhere

Reply:
Ah, I got what you mean. Apologies, I totally missed the point. I looked at how we can make the function accessible. It would not be worth doing this in the Critical CSS module itself since we will be refactoring the class. I think it is worth reverting back to media check inside the `asynchronize_stylesheets` for now and revisiting this with the DOM manipulation logic.

Reply:
I think reverting is going backwards a bit. If you think this is going to get out of hand with refactoring, I think the simplest solution possible is just to make `asynchronize_stylesheets_media` a static method - because it's not reliant upon anything that's going on with the instance.

That would at least allow for code like this:

```php
remove_filter( 'jetpack_boost_async_style', [ Automattic\Jetpack_Boost\Modules\Critical_CSS::class, 'asynchronize_stylesheets_media' ], 10 );
```

But, I'm a bit vary about this as it seems like a band-aid that we don't have a plan for removing.

Reply:
Alternatively - we can just remove the filter as a temporary solution to the problem 😄

Reply:
I agree that it will still be a band-aid if we do not move the filter to the DOM manipulation class. So yes, removed the filter call.